metis-core/metis-core-service/src/test/java/eu/europeana/metis/core/dao/TestDepublishRecordIdDao.java

```diff
@@ -0,0 +1,178 @@
+
+    doReturn(mockDatastore).when(provider).getDatastore();
+    doReturn(mockQuery).when(mockDatastore).find(DepublishRecordId.class);
+    doReturn(1L).when(depublishRecordIdDao).deleteRecords(mockQuery);
+    depublishRecordIdDao.deletePendingRecordIds(datasetId, setTest);
+
+    verify(mockDatastore, times(1)).find(DepublishRecordId.class);
+    verify(mockQuery, times(3)).filter(any());
+    assertEquals(0, provider.getDatastore().find(DepublishRecordId.class).count());
+  }
+
+
+  @Test
+  void countSuccessfullyDepublishedRecordIdsForDatasetTest() {
+    final String datasetId = Integer.toString(TestObjectFactory.DATASETID);
+    final Set<String> setTest = new HashSet<>();
+    setTest.add("1003");
+
+    depublishRecordIdDao
+        .addRecords(setTest, datasetId, DepublicationStatus.DEPUBLISHED, Instant.now());
+    reset(provider);
+    long result = depublishRecordIdDao.countSuccessfullyDepublishedRecordIdsForDataset(datasetId);
+
+    verify(provider, times(1)).getDatastore();
+    assertEquals(1L, result);
+  }
+
+  @Test
+  void getDepublishRecordIdsTest() {
+    final String datasetId = Integer.toString(TestObjectFactory.DATASETID);
+    final Set<String> setTest = new HashSet<>();
+    setTest.add("1004");
+    setTest.add("1005");
+
+    depublishRecordIdDao
+        .addRecords(setTest, datasetId, DepublicationStatus.PENDING_DEPUBLICATION, Instant.now());
+    List<DepublishRecordIdView> find1004 = depublishRecordIdDao
+        .getDepublishRecordIds(datasetId, 0, DepublishRecordIdSortField.DEPUBLICATION_STATE,
+            SortDirection.ASCENDING,
+            "1004");
+
+    List<DepublishRecordIdView> findAll = depublishRecordIdDao
+        .getDepublishRecordIds(datasetId, 0, DepublishRecordIdSortField.DEPUBLICATION_STATE,
+            SortDirection.ASCENDING,
+            null);
+
+    assertEquals(1, find1004.size());
+    assertEquals(2, findAll.size());
+  }
+
+  @Test
+  void getAllDepublishRecordIdsWithStatusTest() throws BadContentException {
+    final String datasetId = Integer.toString(TestObjectFactory.DATASETID);
+    final Set<String> setTest = new HashSet<>();
+    setTest.add("1006");
+
+    depublishRecordIdDao
+        .addRecords(setTest, datasetId, DepublicationStatus.DEPUBLISHED, Instant.now());
+    Set<String> result = depublishRecordIdDao.getAllDepublishRecordIdsWithStatus(datasetId,
+        DepublishRecordIdSortField.DEPUBLICATION_STATE, SortDirection.ASCENDING,
+        DepublicationStatus.DEPUBLISHED, setTest);
+
+    assertEquals(1, result.size());
+  }
+
+  @Test
+  void markRecordIdsWithDepublicationStatusTest() {
+    final String datasetId = Integer.toString(TestObjectFactory.DATASETID);
+    final Set<String> setTest = new HashSet<>();
+    setTest.add("1007");
+
+    depublishRecordIdDao
+        .addRecords(setTest, datasetId, DepublicationStatus.PENDING_DEPUBLICATION, Instant.now());
+    reset(provider, depublishRecordIdDao);
```

Comment:
This shouldn't be necessary!

Reply:
Because we call the method ` depublishRecordIdDao.addRecords`, the verifies later have a misleading number for the parameter `times`. I wanted to check the exact number of times each object and their method are called within the method `depublishRecordIdDao.markRecordIdsWithDepublicationStatus`, and not both. However if there is another way to do it, I would like to know